include/swift/AST/ModuleDependencies.h

```diff
@@ -634,22 +638,45 @@ class ModuleDependencyInfo {
   }
 
   /// Resolve a dependency's set of `imports` with qualified Module IDs
-  void resolveDirectDependencies(const ArrayRef<ModuleDependencyID> dependencyIDs) {
+  void
+  resolveDirectDependencies(const ArrayRef<ModuleDependencyID> dependencyIDs) {
     assert(!storage->resolved && "Resolving an already-resolved dependency");
     storage->resolved = true;
-    storage->resolvedDirectModuleDependencies.assign(dependencyIDs.begin(), dependencyIDs.end());
+    storage->resolvedDirectModuleDependencies.assign(dependencyIDs.begin(),
+                                                     dependencyIDs.end());
   }
 
   /// Set this module's set of Swift Overlay dependencies
-  void setOverlayDependencies(const ArrayRef<ModuleDependencyID> dependencyIDs) {
+  void
+  setOverlayDependencies(const ArrayRef<ModuleDependencyID> dependencyIDs) {
     assert(isSwiftModule());
-    storage->swiftOverlayDependencies.assign(dependencyIDs.begin(), dependencyIDs.end());
+    storage->swiftOverlayDependencies.assign(dependencyIDs.begin(),
+                                             dependencyIDs.end());
   }
 
   const ArrayRef<ModuleDependencyID> getSwiftOverlayDependencies() const {
     return storage->swiftOverlayDependencies;
   }
 
+  const ArrayRef<LinkLibrary> getLinkLibraries() const {
+    return storage->linkLibraries;
+  }
+
+  void
+  setLinkLibraries(const ArrayRef<LinkLibrary> linkLibraries) {
+    storage->linkLibraries.assign(linkLibraries.begin(), linkLibraries.end());
+  }
+
+  bool isStaticLibrary() const {
+    if (auto *detail = getAsSwiftInterfaceModule())
+      return detail->isStatic;
+    else if (auto *detail = getAsSwiftBinaryModule())
+      return detail->isStatic;
+    // TODO: Can Clang module deps indicate they correspond
+    // to a static library?
```

Comment:
No, module maps outside of a framework are only loosely associated with a library via a `link` statement, but that just has the library name and not its type or extension.